tfhe/src/integer/ciphertext/compressed_ciphertext_list.rs

```diff
@@ -195,4 +197,173 @@ mod tests {
 
         assert!(cks.decrypt_bool(&decompressed3));
     }
+
+    const NB_TESTS: usize = 10;
+    const NB_OPERATOR_TESTS: usize = 1;
+    #[test]
+    fn test_ciphertext_compression() {
+        let cks = ClientKey::new(PARAM_MESSAGE_2_CARRY_2_KS_PBS_TUNIFORM_2M64);
+
+        let private_compression_key =
+            cks.new_compression_private_key(COMP_PARAM_MESSAGE_2_CARRY_2_KS_PBS_TUNIFORM_2M64);
+
+        let (compression_key, decompression_key) =
+            cks.new_compression_decompression_keys(&private_compression_key);
+
+        let mut rng = rand::thread_rng();
+
+        let message_modulus: u128 = cks.parameters().message_modulus().0 as u128;
+
+        let num_blocks = 32;
+        for _ in 0..NB_TESTS {
+            // Unsigned
+            let modulus = message_modulus.pow(num_blocks as u32);
+            for _ in 0..NB_OPERATOR_TESTS {
+                let nb_messages = 1 + (rng.gen::<u64>() % 6);
+                let messages = (0..nb_messages)
+                    .map(|_| rng.gen::<u128>() % modulus)
+                    .collect::<Vec<_>>();
+
+                let cts = messages
+                    .iter()
+                    .map(|message| cks.encrypt_radix(*message, num_blocks))
+                    .collect_vec();
```

Comment:
Before compression/decompression we should perform a `bitand` between each message and itself to make sure the noise is the nominal one after a PBS, and not just the encryption noise.

Reply:
I understand this should be at the test and not in compression/decompression algorithms, right?

Reply:
yes it should be in the test to generate integers with post computation levels of noise

so encrypt -> bitand with self -> compress -> decompress -> decrypt -> check decrypted value

Reply:
If I add this line to CPU's these the test will fail every time:

```
                for ct in cts {
                    let and_ct = radix_sks.bitand_parallelized(&ct, &ct);
                    builder.push(and_ct);
                }
```

Reply:
We should probably wait for the fix before doing this, or completely remove this test from the CI in the meantime.

Reply:
The goal here is: pessimize the CI and have the proper setting, wait for the fix and fix here

Reply:
I see. In this case I'm adding the bitand operation before compress.